postreise/analyze/generation/costs.py

```diff
@@ -0,0 +1,59 @@
+import numpy as np
+import pandas as pd
+from numpy.polynomial.polynomial import polyval
+
+from postreise.analyze.check import (
+    _check_gencost,
+    _check_scenario_is_in_analyze_state,
+    _check_time_series,
+)
+
+
+def calculate_costs(
+    scenario=None, pg=None, gencost=None, decommit=False, decommit_threshold=1
+):
+    """Calculate individual generator costs at given powers. If decommit is
+    True, costs will be zero below the decommit threshold (1 MW).
+    Either ``scenario`` XOR (``pg`` AND ``gencost``) must be specified.
+
+    :param powersimdata.scenario.scenario.Scenario scenario: scenario to analyze.
+    :param pandas.DataFrame pg: Generation solution data frame.
+    :param pandas.DataFrame gencost: cost curve polynomials.
+    :param bool decommit: Whether to decommit generator at low power.
+    :param int/float decommit_threshold: The power (MW) below which generators are
+        assumed to be 'decommitted', and costs are zero (if ``decommit`` is True).
+    :raises ValueError: if not (``scenario`` XOR (``pg`` AND ``gencost``)) is specified,
+        or if ``pg`` is passed and has negative values.
+    :return: (*pandas.DataFrame*) -- data frame of costs.
+        Index is hours, columns are plant IDs, values are $/hour.
+    """
+
+    # Check that we've appropriately specified `scenario` XOR (`pg` AND `gencost`)
```

Comment:
Could we use the xor operator? I think this is equivalent (please double check though):
```(scenario is not None) ^ (pg is not None and gencost is not None)```

Reply:
Nice! Done.